core/services/workflows/syncer/workflow_registry.go

```diff
@@ -362,6 +355,192 @@ func (w *workflowRegistry) readRegistryEvents(ctx context.Context, don capabilit
+				OldWorkflowID: engine.workflowID,
+				NewWorkflowID: wfMeta.WorkflowID,
+				WorkflowOwner: wfMeta.Owner,
+				DonID:         wfMeta.DonID,
+				WorkflowName:  wfMeta.WorkflowName,
+				BinaryURL:     wfMeta.BinaryURL,
+				ConfigURL:     wfMeta.ConfigURL,
+				SecretsURL:    wfMeta.SecretsURL,
+			}
+			err := w.handler.Handle(ctx, workflowAsEvent{
+				Data:      toRegisteredEvent,
+				EventType: WorkflowUpdatedEvent,
+			})
+			if err != nil {
+				w.lggr.Errorw("failed to handle event", "err", err, "type", WorkflowUpdatedEvent)
+				continue
+			}
+
+			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
+			continue
+		}
+
+		// if the workflow engine is running, the metadata is the same, and the status is active
+		// then the workflow is unchanged. mark as seen.
+		if wfMeta.Status == 1 {
+			engineKeyMap[wfMeta.WorkflowName+hex.EncodeToString(wfMeta.Owner)] = true
+			continue
+		}
+
+		// if the workflow engine is running, the metadata is the same, but the status isn't active
+		// then the workflow is has been paused. NOOP to be handled below as a deleted event, which clears the DB workflow spec.
+	}
+
+	// Shut down engines that are no longer in the contract's latest workflow metadata state
+	allEngines := w.engineRegistry.GetAll()
+	for _, engine := range allEngines {
+		keyStr := engine.workflowName + hex.EncodeToString(engine.workflowOwner)
+		_, exists := engineKeyMap[keyStr]
+
+		if !exists {
+			toDeletedEvent := WorkflowRegistryWorkflowDeletedV1{
+				WorkflowID:    engine.workflowID,
+				WorkflowOwner: engine.workflowOwner,
+				DonID:         donID,
+				WorkflowName:  engine.workflowName,
+			}
+			err := w.handler.Handle(ctx, workflowAsEvent{
+				Data:      toDeletedEvent,
+				EventType: WorkflowDeletedEvent,
+			})
+			if err != nil {
+				w.lggr.Errorw("failed to handle event", "err", err, "type", WorkflowDeletedEvent)
+			}
+		}
+	}
+}
+
+// pollRegistryState polls every w.ticker tick for the latest workflow state and
+// calls the handler to bring the system's state up to date with the difference.
+func (w *workflowRegistry) pollRegistryStateToEvents(ctx context.Context, don capabilities.DON, reader ContractReader) {
+	ticker := w.getTicker()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker:
+			workflowMetadata, _, err := w.getWorkflowMetadata(ctx, don, reader)
+			if err != nil {
+				w.lggr.Errorw("failed to process registry state", "err", err)
+				continue
+			}
+			w.handleWorkflowMetadataAsEvents(ctx, workflowMetadata, don.ID)
+		}
+	}
+}
+
+// syncUsingReconciliationStrategy syncs workflow registry contract state by polling the workflow metadata state and comparing to local state.
+// It still watches for ForceUpdateSecretsEvents, which can't be reconciled through workflow metadata state.
+func (w *workflowRegistry) syncUsingReconciliationStrategy(ctx context.Context, don capabilities.DON, reader ContractReader) {
```

Comment:
Style nit -- you've abstracted too much out here; there's probably some parts of the procedure that you want to un-abstract and pull back up into this